### ISSUE TITLE
Config dump format updated, breaking `istioctl pc`

### ISSUE THREAD

**Issue by howardjohn** [istio/istio#15544]
Steps to reproduce:

* Install istio from master
* `istioctl proxy config bootstrap POD`

Error:
`Error: error unmarshalling config dump response from Envoy: unknown message type "envoy.admin.v2alpha.ScopedRoutesConfigDump"`

Config dump shows some new field?
```
  {
   "@type": "type.googleapis.com/envoy.admin.v2alpha.ScopedRoutesConfigDump"
  },
```

This probably breaks some other things too

fyi @kyessenov @esnible

**Comment by kyessenov** [comment]
Well, it's `v2alpha` API. Please don't use it for anything serious.

**Comment by howardjohn** [comment]
Actually this fails on istioctl 1.2.2 but seems to work from HEAD (e4a3f6b974a7f7bb6b7c3866051b3555ada0b3b4). I think that since we have pulled in the new go-control-plane this works but we no longer have forward compatibility

**Comment by esnible** [comment]
@howardjohn Should I be trying to fix this?  I am unclear which versions conflict.  Let me know if work is needed for 1.3, and if you recommend an implementation strategy of being more tolerant deserializing JSON, or switching to different Envoy structure versions.

**Comment by howardjohn** [comment]
if we can make it ignore unknown fields that may be sufficient to get 1.2 istioctl working with proxy 1.3. Otherwise istioctl 1.2 will fail for any proxy running 1.3.

**Comment by howardjohn** [comment]
Fixed in #15601

**Comment by elvizlai** [comment]
still failed in istio v1.4.3
```
istioctl proxy-status istio-ingressgateway-66c48cfd5d-9bgt7.istio-system

Error: unknown message type "envoy.admin.v2alpha.ScopedRoutesConfigDump"
```

```
kubectl describe pod istio-ingressgateway-66c48cfd5d-9bgt7 -n istio-system

Name:         istio-ingressgateway-66c48cfd5d-9bgt7
Namespace:    istio-system
Priority:     0
Node:         k8s-node03/172.16.0.29
Start Time:   Thu, 09 Jan 2020 09:09:11 +0800
Labels:       app=istio-ingressgateway
              chart=gateways
              heritage=Tiller
              istio=ingressgateway
              pod-template-hash=66c48cfd5d
              release=istio
Annotations:  cni.projectcalico.org/podIP: 10.244.135.162/32
              kubectl.kubernetes.io/restartedAt: 2019-09-23T10:06:43+08:00
              sidecar.istio.io/inject: false
Status:       Running
IP:           10.244.135.162
IPs:
  IP:           10.244.135.162
Controlled By:  ReplicaSet/istio-ingressgateway-66c48cfd5d
Containers:
  ingress-sds:
    Container ID:   docker://d6719c177641b5e9731685256f7d0b1f31e95d70297fce47bb6625d9c1b105d2
    Image:          docker.io/istio/node-agent-k8s:1.4.3
    Image ID:       docker-pullable://istio/node-agent-k8s@sha256:976d435231f9822a958301b06d8c21c238753638b366341cd7094cdba8794fc0
    Port:           <none>
    Host Port:      <none>
    State:          Running
      Started:      Thu, 09 Jan 2020 09:09:51 +0800
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     2
      memory:  1Gi
    Requests:
      cpu:     50m
      memory:  50Mi
    Environment:
      ENABLE_WORKLOAD_SDS:         false
      ENABLE_INGRESS_GATEWAY_SDS:  true
      INGRESS_GATEWAY_NAMESPACE:   istio-system (v1:metadata.namespace)
    Mounts:
      /var/run/ingress_gateway from ingressgatewaysdsudspath (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from istio-ingressgateway-service-account-token-nm2sj (ro)
  istio-proxy:
    Container ID:  docker://b39e09086642fb05647d9aee71ce714e838f646dabf24fa3a956691ffecb3770
    Image:         docker.io/istio/proxyv2:1.4.3
    Image ID:      docker-pullable://istio/proxyv2@sha256:ca781f35002eecb6cd1b4dd359e4783cddaac3a8df0bc6e6f966e01637c93e86
    Ports:         80/TCP, 443/TCP, 15011/TCP, 8060/TCP, 853/TCP, 15090/TCP
    Host Ports:    0/TCP, 0/TCP, 0/TCP, 0/TCP, 0/TCP, 0/TCP
    Args:
      proxy
      router
      --domain
      $(POD_NAMESPACE).svc.cluster.local
      --proxyLogLevel=warning
      --proxyComponentLogLevel=misc:error
      --log_output_level=default:info
      --drainDuration
      45s
      --parentShutdownDuration
      1m0s
      --connectTimeout
      10s
      --serviceCluster
      istio-ingressgateway
      --zipkinAddress
      zipkin.istio-system:9411
      --proxyAdminPort
      15000
      --statusPort
      15020
      --controlPlaneAuthPolicy
      MUTUAL_TLS
      --discoveryAddress
      istio-pilot.istio-system:15011
      --trust-domain=cluster.local
    State:          Running
      Started:      Thu, 09 Jan 2020 09:10:43 +0800
    Ready:          True
    Restart Count:  0
    Limits:
      cpu:     2
      memory:  1Gi
    Requests:
      cpu:      100m
      memory:   128Mi
    Readiness:  http-get http://:15020/healthz/ready delay=1s timeout=1s period=2s #success=1 #failure=30
    Environment:
      aliyun_logs_envoy:            stdout
      aliyun_logs_envoy_tags:       fields.topic=log:envoy
      NODE_NAME:                     (v1:spec.nodeName)
      POD_NAME:                     istio-ingressgateway-66c48cfd5d-9bgt7 (v1:metadata.name)
      POD_NAMESPACE:                istio-system (v1:metadata.namespace)
      INSTANCE_IP:                   (v1:status.podIP)
      HOST_IP:                       (v1:status.hostIP)
      SERVICE_ACCOUNT:               (v1:spec.serviceAccountName)
      ISTIO_META_WORKLOAD_NAME:     istio-ingressgateway
      ISTIO_META_OWNER:             kubernetes://apis/apps/v1/namespaces/istio-system/deployments/istio-ingressgateway
      ISTIO_META_MESH_ID:           cluster.local
      ISTIO_META_POD_NAME:          istio-ingressgateway-66c48cfd5d-9bgt7 (v1:metadata.name)
      ISTIO_META_CONFIG_NAMESPACE:  istio-system (v1:metadata.namespace)
      ISTIO_META_USER_SDS:          true
      ISTIO_META_ROUTER_MODE:       sni-dnat
      ISTIO_METAJSON_LABELS:        {"app":"istio-ingressgateway","istio":"ingressgateway"}

      ISTIO_META_CLUSTER_ID:        Kubernetes
      SDS_ENABLED:                  false
    Mounts:
      /etc/certs from istio-certs (ro)
      /etc/istio/ingressgateway-ca-certs from ingressgateway-ca-certs (ro)
      /etc/istio/ingressgateway-certs from ingressgateway-certs (ro)
      /var/run/ingress_gateway from ingressgatewaysdsudspath (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from istio-ingressgateway-service-account-token-nm2sj (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             True
  ContainersReady   True
  PodScheduled      True
Volumes:
  ingressgatewaysdsudspath:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  istio-certs:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  istio.istio-ingressgateway-service-account
    Optional:    true
  ingressgateway-certs:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  istio-ingressgateway-certs
    Optional:    true
  ingressgateway-ca-certs:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  istio-ingressgateway-ca-certs
    Optional:    true
  istio-ingressgateway-service-account-token-nm2sj:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  istio-ingressgateway-service-account-token-nm2sj
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:          <none>
```
